runners/direct-java/src/main/java/org/apache/beam/runners/direct/WatermarkManager.java

```diff
@@ -366,32 +365,32 @@ synchronized void updateTimers(TimerUpdate update) {
             keyTimers.add(timer);
           }
 
-          existingTimersForKey.put(
-              timer.getNamespace(), timer.getTimerId() + '+' + timer.getTimerFamilyId(), timer);
+          existingTimersForKey.put(timer.getNamespace(), timerKey, timer);
         }
       }
 
       for (TimerData timer : update.getDeletedTimers()) {
         if (TimeDomain.EVENT_TIME.equals(timer.getDomain())) {
+          String timerKey = timer.getTimerId() + '+' + timer.getTimerFamilyId();
```

Comment:
The construction of the `timerKey` seems repeated, could we wrap it in a helper method?

Reply:
Cleaned up timerKey usage